lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalMatches.java

```diff
@@ -74,7 +75,7 @@ public MatchesIterator getSubMatches() throws IOException {
 
       @Override
       public Query getQuery() {
-        throw new UnsupportedOperationException();
+        return query==null ? query = new IntervalQuery(field, intervalsSource) : query;
```

Comment:
This is a little bit doubtful. WDYT, @romseygeek ?

Reply:
```suggestion
        return source.getQuery();
```

Then we don't need pass the field and source around everywhere